tensorflow/lite/micro/benchmarks/person_detection_benchmark.cc

```diff
@@ -43,48 +43,65 @@ alignas(16) uint8_t tensor_arena[kTensorArenaSize];
 
 uint8_t op_resolver_buffer[sizeof(PersonDetectionOpResolver)];
 uint8_t benchmark_runner_buffer[sizeof(PersonDetectionBenchmarkRunner)];
-PersonDetectionBenchmarkRunner* benchmark_runner = nullptr;
 
 // Initialize benchmark runner instance explicitly to avoid global init order
 // issues on Sparkfun. Use new since static variables within a method
 // are automatically surrounded by locking, which breaks bluepill and stm32f4.
-void CreateBenchmarkRunner() {
+PersonDetectionBenchmarkRunner* CreateBenchmarkRunner(MicroProfiler* profiler) {
   // We allocate PersonDetectionOpResolver from a global buffer
   // because the object's lifetime must exceed that of the
   // PersonDetectionBenchmarkRunner object.
-  benchmark_runner = new (benchmark_runner_buffer)
-      PersonDetectionBenchmarkRunner(g_person_detect_model_data,
-                                     new (op_resolver_buffer)
-                                         PersonDetectionOpResolver(),
-                                     tensor_arena, kTensorArenaSize);
+  return new (benchmark_runner_buffer) PersonDetectionBenchmarkRunner(
+      g_person_detect_model_data,
+      new (op_resolver_buffer) PersonDetectionOpResolver(), tensor_arena,
+      kTensorArenaSize, profiler);
 }
 
-void InitializeBenchmarkRunner() {
-  CreateBenchmarkRunner();
-  benchmark_runner->SetInput(reinterpret_cast<const int8_t*>(g_person_data));
-}
-
-void PersonDetectionTenIerationsWithPerson() {
-  benchmark_runner->SetInput(reinterpret_cast<const int8_t*>(g_person_data));
-  for (int i = 0; i < 10; i++) {
-    benchmark_runner->RunSingleIteration();
+void PersonDetectionNIerations(const int8_t* input, int iterations,
+                               const char* tag,
+                               PersonDetectionBenchmarkRunner& benchmark_runner,
+                               MicroProfiler& profiler) {
+  benchmark_runner.SetInput(input);
+  int32_t ticks = 0;
+  for (int i = 0; i < iterations; ++i) {
+    profiler.ClearEvents();
+    benchmark_runner.RunSingleIteration();
+    ticks += profiler.GetTotalTicks();
   }
+  MicroPrintf("%s took %d ticks (%d ms)", tag, ticks, TicksToMs(ticks));
 }
 
-void PersonDetectionTenIerationsWithoutPerson() {
-  benchmark_runner->SetInput(reinterpret_cast<const int8_t*>(g_no_person_data));
-  for (int i = 0; i < 10; i++) {
-    benchmark_runner->RunSingleIteration();
-  }
+}  // namespace tflite
+
+int main(int argc, char** argv) {
+  tflite::MicroProfiler profiler;
+
+  uint32_t event_handle = profiler.BeginEvent("InitializeBenchmarkRunner");
+  tflite::PersonDetectionBenchmarkRunner* benchmark_runner =
+      CreateBenchmarkRunner(&profiler);
+  profiler.EndEvent(event_handle);
+  profiler.Log();
+  MicroPrintf("");
```

Comment:
Same question about these printfs. Are they necessary for flushing or something? Perhaps a comment if they are necessary.

Reply:
added a comment.